src/array/fixed_size_binary/mod.rs

```diff
@@ -136,6 +136,29 @@ impl FixedSizeBinaryArray {
             .get_unchecked(i * self.size..(i + 1) * self.size)
     }
 
+    /// Returns a new [`FixedSizeBinary`] with a different logical type.
+    /// This is `O(1)`.
+    /// # Panics
+    /// Panics iff the data_type is not supported for the physical type.
+    #[inline]
+    pub fn to(self, data_type: DataType) -> Self {
+        match (
```

Comment:
Something like this? I assume we must check that the physical types are correct.

Reply:
exactly, this works!